src/Riok.Mapperly/Descriptors/DescriptorBuilder.cs

```diff
@@ -123,9 +127,29 @@ private void ExtractUserMappings()
                 userMapping.TargetType
             );
 
+            if (userMapping.Method is { IsStatic: true, IsPartialDefinition: true })
+            {
+                _mapperDescriptor.Static = true;
+            }
+            else if (firstNonStaticUserMapping is null && userMapping.Method is { IsStatic: false, MethodKind: not MethodKind.Constructor })
```

Comment:
Is the second part of the condition really required? This can't ever be a constructor, can it? `ExtractUserMappings` should never return a ctor. Also if the if condition is simplified to `userMapping.Method.IsStatic` the else if could be simplified to `firstNonStaticUserMapping == null` since the static part is already ensured to be false by the if condition.

Reply:
You were right with ctor, the condition was left there from moment where I used `ExtractMethods` directly

Reply:
I think the condition cannot be simplified due to `InstanceMapperShouldSupportUserDefinedStaticMethods` mention above